septentrion/versions.py

```diff
@@ -22,3 +22,9 @@ def __eq__(self, other: object) -> bool:
     def __repr__(self) -> str:
         version_string = ".".join(str(n) for n in self._version)
         return f'{self.__class__.__name__}("{version_string}")'
+
+    def __hash__(self) -> int:
+        return hash(self._version)
+
+    def normalized_repr(self) -> str:
+        return ".".join(str(i) for i in self._version)
```

Comment:
Does this mean we cannot use "19.04" as version number ?

I think I'd be more comfortable if the version number original string was kept as an attribute of the object ?

Reply:
Ok, the more I see it, the more a dataclass makes sense.

Even if it means we'll need a dedicated constructor `Version.from_string()`